server/static/js/index/home_page_modal_windows.js

```diff
@@ -0,0 +1,293 @@
+            $(MODAL_USER_LAST_NAME).attr("placeholder","Edit Last Name");
+            $(MODAL_USER_EMAIL).attr("placeholder","Edit email");
+            $(MODAL_USER_PASSWORD).attr("placeholder","Edit password");
+            $(MODAL_USER_CONFIRM_PASSWORD).attr("placeholder","Confirm password");
+            openModalAnimation();
+            $('#btn_modal_user_edit').css(DISPLAY_BLOCK);
+            $(MODAL_USER).css(DISPLAY_FLEX);
+        }
+    });
+
+    // Close device modal windows. With zeroing output
+    function closeModalDevice () {
+        closeModalAnimation();
+        setTimeout(function () {
+           $(MODAL_DEVICE).css(DISPLAY_NONE);
+           $(MODAL_DEVICE_ID).val("");
+           $(MODAL_DEVICE_KEY).val("");
+           $(MODAL_DEVICE_TYPE).val("1");
+           $('[id *= btn_modal]').css(DISPLAY_NONE);
+        }, setTime);
+    }
+
+    // Close user modal windows. With zeroing output
+    function closeModalUser () {
+        closeModalAnimation();
+        setTimeout(function () {
+           $(MODAL_USER).css(DISPLAY_NONE);
+           $(MODAL_USER_FIRST_NAME).val("");
+           $(MODAL_USER_LAST_NAME).val("");
+           $(MODAL_USER_EMAIL).val("");
+           $(MODAL_USER_PASSWORD).val("");
+           $(MODAL_USER_CONFIRM_PASSWORD).val("");
+           $('#modal_user_active').prop("checked", false);
+           $(MODAL_USER_CHECKBOX).each(function () {
+               $(this).prop('checked', false);
+           });
+           $('[id *= btn_modal]').css(DISPLAY_NONE);
+        }, setTime);
+    }
+
+    $(CLOSE_MODAL).click(function () {
+        if ($(this).data("modal") === "device") {
+            closeModalDevice();
+        } else {
+            closeModalUser();
+        }
+    });
+
+    // Device editing function followed by an AJAX request
+    $(BTN_MODAL_DEVICE_EDIT).click(function () {
+        let payload = {"deviceData": {}};
+        closeModalDevice();
+        sendJSONRequest("/home/device.svc", updateDevice(payload), RequestMethod.PUT, beforeLoadTableDevice,
+                        renderTableDevices, debug_callback, process_failures);
+    });
+
+    // Creating an AJAX request to add a new device
+    $(BTN_MODAL_DEVICE_ADD).click(function () {
+        let payload = {"deviceData": {}};
+        closeModalDevice();
+        sendJSONRequest("/home/device.svc", createDevice(payload), RequestMethod.POST, beforeLoadTableDevice,
+                        renderTableDevices, debug_callback, process_failures);
+    });
+
+    $(BTN_MODAL_USER_EDIT).click(function () {
+        let payload = {"userData": {"roles":[]}};
+        closeModalUser();
+        sendJSONRequest("/home/user.svc", updateUser(payload), RequestMethod.PUT, beforeLoadTableUsers,
+                        renderTableUsers, debug_callback, process_failures);
+    });
+
+    $(BTN_MODAL_USER_ADD).click(function () {
+        let payload = {"userData": {"roles":[]}};
```

Comment:
Seems like initiating the payload in event-loop makes no sense - not only we duplicate the code, we make a meaningless argument pass when we call payload population function.

Consider moving the payload initialization in respective payload processing function (getUserPayload for devices)